src/Storage/Controllers/InstancesController.cs

```diff
@@ -118,13 +118,59 @@ public async Task<ActionResult<QueryResponse<Instance>>> GetInstances(InstanceQu
                 {
                     queryParameters.Org = queryParameters.AppId.Split('/')[0];
                 }
-
+                
                 if (!orgClaim.Equals(queryParameters.Org, StringComparison.InvariantCultureIgnoreCase))
                 {
-                    return Forbid();
-                }
+                    var appId = ValidateAppId(queryParameters.AppId).Split("/")[1];
+                    XacmlJsonRequestRoot request;
+                    try
+                    {
+                        var instanceOwnerPartyId = (int)queryParameters.InstanceOwnerPartyId;
+                        request = DecisionHelper.CreateDecisionRequest(queryParameters.Org, appId, HttpContext.User, "read", instanceOwnerPartyId, null);
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.LogError(ex, $"Something went wrong during CreateDecisionRequest for application id: {appId}");
```

Comment:
This type of logging followed by a throw should be unnecessary.

If we need to add information to an exception, like a better message, then create a new exception with situational specific message and properties, the original exception can be provided as an inner exception.

Logging would be required if we didn't have the throw.

Reply:
Alright. So it should be fine to just remove these log statements? It will be handled by the throwing?

Reply:
In that case it would be better to remove the try catch completely. There are exception handling in the middleware, call chain, and request pipeline. 

However, any unhandled exception will be logged and converted to a 500 internal server error response. Is that what we want here?

Reply:
Hm. I guess if any of these try-catches actually throw an error, the application haven't been able to determine whether or not the request should have access. I think any sort of 4xx might be confusing for the caller. It would be better if it returned 500 internal server error. 

What do you think?